test/cram/test.t

```diff
@@ -21,28 +21,28 @@ This is a cram test for the new executable.
   
   
   
-  Type 'h' to go to the help panel, 'q' to quit, 'n' to go to the next operation, 'p' to go to the previous operation. Press 't' to toggle view mode.
+  [h]elp [q]uit [n/p]avigate [t]oggle view [r]ender mode
   $ dummy_terminal example.diff n
   Operation 2 of 14, 1 hunk
   3 additions, 1 removal
   Modification of bin/main.ml
   @@ -1,7 +1,9 @@
-   1  1   open Diffcessible
-   2  2   
-   3  3   let main () =
-   4    -   Interactive_viewer.start ()
-      4 +   let s = In_channel.input_all In_channel.stdin in
-      5 +   let patch = Patch.to_diffs s in
-      6 +   Interactive_viewer.start patch
-   5  7   
-   6  8   open Cmdliner
-   7  9   
+   1  1   open Diffcessible 
+   2  2    
+   3  3   let main () = 
+   4    -   Interactive_viewer.start () 
+      4 +   let s = In_channel.input_all In_channel.stdin in   let patch = Patch.to_diffs s in   Interactive_viewer.start patch 
```

Comment:
This is not right, two lines are merged together. On main we have a check that avoids doing word diff in this case but it's removed in this PR.

Reply:
Resolved. Reverted cram tests to follow main

Reply:
Reverting the cram tests don't make sense, you need to run and promote them. If I run them, this bug appear again, so something is broken in the code.

Reply:
Got it

Reply:
To be clear, on `main` this renders as:

```
 1  1   open Diffcessible
 2  2   
 3  3   let main () =
 4    -   Interactive_viewer.start ()
    4 +   let s = In_channel.input_all In_channel.stdin in
    5 +   let patch = Patch.to_diffs s in
    6 +   Interactive_viewer.start patch
 5  7   
 6  8   open Cmdliner
 7  9   
```

With this PR, this now renders like this:

```
 1  1   open Diffcessible 
 2  2    
 3  3   let main () = 
 4    -   Interactive_viewer.start () 
    4 +   let s = In_channel.input_all In_channel.stdin in   let patch = Patch.to_diffs s in   Interactive_viewer.start patch 
 5  5    
 6  6   open Cmdliner 
 7  7    
```

This seems to be related to `render_hunk`. In `ui_unified_diff`, there's a check whether or not to enable word diffing. In the false case, `render_hunk` gets called, which compute word diffs anyway.
That might be a mistake during a refactoring. By the way, `render_hunk` has a clear purpose and might have a more descriptive name.

Reply:
Fixed